compiler/rustc_typeck/src/check/expr.rs

```diff
@@ -1974,7 +1974,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
 
             field_path.push(candidate_field.ident.normalize_to_macros_2_0());
             let field_ty = candidate_field.ty(self.tcx, subst);
-            if let Some((nested_fields, _)) = self.get_field_candidates(span, &field_ty) {
+            if let Some((nested_fields, subst)) = self.get_field_candidates(span, &field_ty) {
```

Comment:
Random note about something I noticed, that I don't think is worth touching in this PR: this function used `subst` instead of `substs`, and while we do use the name `subst`, it's for the method (where it means "substitute") - the variables of type `Substs` ("substitutions") should be called `substs`.